### PR TITLE
Swap all occurrences of `.then()` `.catch()` `.finally()` with try/catch statements

### PR DESCRIPTION
**Pull request by imcdonald-d2l**
Swapping all occurrences of `.then()` `.catch()` `.finally()` with try catch statements. This allows us to implement `async` functions which we can `await` in unit tests so that we can ensure code has executed as expected.

**Note:** having `async` functions with `try/catch` is the pattern we should be adopting moving forward so that we can can better control things in unit tests. 

**Caveat**: not replacing `.then` references in [entity-watcher-mixin](https://github.com/Brightspace/d2l-rubric/blob/master/src/mixins/entity-watcher-mixin.js#L32) as they are more complex uses that shouldn't be necessary to fix for unit tests 🤞🏻

### CODE REVIEW

File: src/components/editor/rubric-criteria-editor.js [hunk 1]
```diff
@@ -103,48 +103,48 @@ class RubricCriteriaEditor extends composeMixins(
...
+			} catch (err) {
 				this.dispatchEvent(new CustomEvent('d2l-rubric-editor-save-error', {
 					detail: {
 						message: err.message,
 					},
 					bubbles: true,
 					composed: true,
 				}));
-			}).then(() => {
+			} finally {
```
Comment (devpow112) [inline src/components/editor/rubric-criteria-editor.js:117]: Why the change to finally here?

File: src/components/editor/rubric-criteria-group-editor.js [hunk 2]
```diff
@@ -273,27 +275,27 @@ class RubricCriteriaGroupEditor extends composeMixins(
...
+			} catch (err) {
 				this.dispatchEvent(new CustomEvent('d2l-rubric-editor-save-error', {
 					detail: {
 						message: err.message,
 					},
 					bubbles: true,
 					composed: true,
 				}));
-			}).then(() => {
+			} finally {
```
Comment (devpow112) [inline src/components/editor/rubric-criteria-group-editor.js:286]: Why the change to finally here?

File: src/components/rubric-editable-score.js [hunk 3]
```diff
@@ -505,75 +505,60 @@ class RubricEditableScore extends composeMixins(
...
 				} else {
 					e.target.value = this.score;
 					return;
 				}
 			}
-
-			this._scoreSaves++;
-			this.score = newScore;
-
-			action.catch(error => {
```
Comment (devpow112) [inline src/components/rubric-editable-score.js:566]: What happened to this code?

File: src/components/editor/rubric-criteria-editor.js (hunk 1 above)
Comment (imcdonald-d2l) [reply src/components/editor/rubric-criteria-editor.js:117]: Isn't necessary. Good catch.

File: src/components/editor/rubric-criteria-group-editor.js (hunk 2 above)
Comment (imcdonald-d2l) [reply src/components/editor/rubric-criteria-group-editor.js:286]: Also not necessary 👍🏻

File: src/components/editor/rubric-level-editor.js [hunk 4]
```diff
@@ -363,24 +366,25 @@ class RubricLevelEditor extends composeMixins(
...
 						bubbles: true,
 						composed: true,
 					}));
 					this._updateName(this.entity);
-				}).catch((err) => {
+				} catch (err) {
 					this._nameInvalid = true;
 					this._nameInvalidError = tryGetSirenErrorMessage(err) || this.localize('rubricEditor:nameSaveFailed');
-				}).then(() => {
+				} finally {
```
[the remark this answers is not in the document]
Comment (imcdonald-d2l) [reply src/components/editor/rubric-level-editor.js:379]: Not necessary, will fix 👍🏻

File: src/components/rubric-editable-score.js (hunk 3 above)
Comment (imcdonald-d2l) [reply src/components/rubric-editable-score.js:566]: This code got moved to `_handleScoreFinally` and `_handleScoreSaveError` below. They are executed on the actions in `this._saveScore(newScore)` and `this._clearOverride()`.

Comment (devpow112) [reply src/components/rubric-editable-score.js:566]: Ah I see. Ok cool. Thanks for the explanation.
